addons/web/static/src/js/libs/fullcalendar.js

```diff
@@ -0,0 +1,214 @@
+            Calendar,
+            createElement,
+            EventApi,
+            memoizeRendering,
+            View,
+        } = FullCalendar;
+
+        class YearView extends View {
+            constructor() {
+                super(...arguments);
+                this.months = null;
+                this.renderSubCalendarsMem = memoizeRendering(
+                    this.renderSubCalendars, this.unrenderSubCalendars);
+                this.events = [];
+            }
+
+            //----------------------------------------------------------------------
+            // Getters
+            //----------------------------------------------------------------------
+
+            get currentDate() {
+                return this.context.calendar.state.currentDate;
+            }
+
+            //----------------------------------------------------------------------
+            // Public
+            //----------------------------------------------------------------------
+
+            /**
+             * @override
+             */
+            destroy() {
+                this.renderSubCalendarsMem.unrender();
+                super.destroy();
+            }
+            /**
+             * Removes the selection on sub calendar.
+             * Selections on sub calendars are not propagated to this view so
+             * this view cannot manage them.
+             */
+            unselect() {
+                for (const { calendar } of this.months) {
+                    calendar.unselect();
+                }
+            }
+            /**
+             * @override
+             */
+            render() {
+                this.renderSubCalendarsMem(this.context);
+                super.render(...arguments);
+            }
+            /**
+             * Renders the main layout (the 4x3 month grid)
+             */
+            renderSubCalendars() {
+                this.el.classList.add('fc-scroller');
+                if (!this.context.options.selectable) {
+                    this.el.classList.add('fc-readonly-year-view');
+                }
+                this.months = [];
+                for (let monthNumber = 0; monthNumber < 12; monthNumber++) {
+                    const monthDate = new Date(this.currentDate.getFullYear(), monthNumber);
+                    const monthShortName = moment(monthDate).format('MMM').toLowerCase();
+                    const container = createElement('div', { class: 'fc-month-container' });
+                    this.el.appendChild(container);
+                    const el = createElement('div', {
+                        class: `fc-month fc-month-${monthShortName}`,
+                    });
+                    container.appendChild(el);
+                    const calendar = this._createMonthCalendar(el, monthDate);
+                    this.months.push({ el, calendar });
+                    calendar.render();
+                }
+            }
+            unrenderSubCalendars() {
```

Comment:
Missing doc. Would be nice to have some explanations on what it does and when it is called.